#readingDataPointsFilters:
#  - RandomSamplingDataPointsFilter:
#      prob: 0.75
readingDataPointsFilters:
  - RandomSamplingDataPointsFilter:
      prob: 0.8

referenceDataPointsFilters:
  - SamplingSurfaceNormalDataPointsFilter:
      knn: 10

matcher:
  KDTreeMatcher:
    knn: 1
    epsilon: 0 

outlierFilters:
  - MaxDistOutlierFilter:
      maxDist: 0.1
#  - TrimmedDistOutlierFilter:
#      ratio: 0.8

errorMinimizer:
  PointToPlaneErrorMinimizer

transformationCheckers:
  - CounterTransformationChecker:
      maxIterationCount: 100
  - DifferentialTransformationChecker:
      minDiffRotErr: 0.001
      minDiffTransErr: 0.01
      smoothLength: 4   

inspector:
  NullInspector
#  PerformanceInspector:
#    baseFileName: performance.txt
#    dumpPerfOnExit: 1
#    dumpStats: 1
#  VTKFileInspector:
#     baseFileName: pointmatcher-run1
#     dumpPerfOnExit: 0
#     dumpStats: 0
#     dumpIterationInfo: 1 
#     dumpDataLinks: 0 
#     dumpReading: 0 
#     dumpReference: 0 


logger:
  NullLogger


